lib/util.ex

```diff
@@ -75,31 +73,40 @@ defmodule Util do
 
     case HTTPoison.post("https://api.start.gg/gql/alpha", JSON.encode!(body), headers) do
       {:ok, resp} ->
-        case JSON.decode(resp.body) do
-          {:ok, response = %{"success" => false}} ->
-            IO.puts(Map.get(response, "message"))
-          {:ok, response} ->
-            response
-            |> Map.get("data")
-            |> Map.get("tournaments")
-            |> Map.get("nodes")
-            # Finesse data into our own non-start.gg format.
-            |> Enum.map(fn sgg_data ->
-              %{
-                "external_id" => "smashgg-" <> Integer.to_string(sgg_data["id"]),
-                "name" => sgg_data["name"],
-                "location" => %{
-                  "lat" => sgg_data["lat"],
-                  "lng" => sgg_data["lng"]
-                },
-                "start_time" => sgg_data["startAt"],
-                "end_time" => sgg_data["endAt"],
-                "updated_at" => sgg_data["updatedAt"],
-                "url" => "https://start.gg/" <> (sgg_data["slug"])
-              }
-              end )
-        end
-      {:error, reason} -> IO.puts(reason)
+        resp.body
+        |> JSON.decode
+        |> format_decoded_response
+      {:error, reason} ->
+        IO.puts(:stderr, reason)
+        System.stop(1)
+    end
+  end
+
+  def format_decoded_response({:ok, decoded_map}) do
+    case {:ok, decoded_map} do
```

Comment:
There's not much point in doing case on a literal (in this case the `:ok` bit), since the `:ok` will by definition always be there. I'd say let's either case on `decoded_map`, or accept a more broad `decoded_response` and case on that.